Compilation fails with triangle

**Description**
When building in Ubuntu

**Scope**
- Compilation
- KratosCore
- Triangle

**To Reproduce**
Build with `-DUSE_TRIANGLE_NONFREE_TPL=ON` and `-DUSE_TETGEN_NONFREE_TPL=ON`

**Console output**
```
-- Build files have been written to: /home/javi/Projects/Kratos/build/Release
[  1%] Built target triangle
[  1%] Built target tinyexpr
[  1%] Built target KratosVersion
[  2%] Built target gidpost
[  2%] Built target runkratos
[  2%] Built target tetgen
[  2%] Building CXX object kratos/CMakeFiles/KratosCore.dir/modeler/cad_tessellation_modeler.cpp.o
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp: In member function ‘double Kratos::CadTessellationModeler::ComputeDiscretizationError(const std::vector<boost::numeric::ublas::bounded_matrix<double, 3, 3, boost::numeric::ublas::basic_row_major<> > >&, const std::vector<boost::numeric::ublas::bounded_matrix<double, 3, 3, boost::numeric::ublas::basic_row_major<> > >&)’:
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:336:83: error: ‘MaxReduction’ was not declared in this scope
  336 |     double discretization_error = IndexPartition<IndexType>(n_triangles).for_each<MaxReduction<double>>([&](IndexType iTriangle){
      |                                                                                   ^~~~~~~~~~~~
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:336:74: error: parse error in template argument list
  336 |     double discretization_error = IndexPartition<IndexType>(n_triangles).for_each<MaxReduction<double>>([&](IndexType iTriangle){
      |                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:351:6: error: no matching function for call to ‘Kratos::IndexPartition<long unsigned int>::for_each<<expression error> >(Kratos::CadTessellationModeler::ComputeDiscretizationError(const std::vector<boost::numeric::ublas::bounded_matrix<double, 3, 3, boost::numeric::ublas::basic_row_major<> > >&, const std::vector<boost::numeric::ublas::bounded_matrix<double, 3, 3, boost::numeric::ublas::basic_row_major<> > >&)::<lambda(Kratos::CadTessellationModeler::IndexType)>)’
  351 |     });
      |      ^
In file included from /home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:17:
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:374:17: note: candidate: ‘template<class TUnaryFunction> void Kratos::IndexPartition<TIndexType, TMaxThreads>::for_each(TUnaryFunction&&) [with TUnaryFunction = TUnaryFunction; TIndexType = long unsigned int; int TMaxThreads = 128]’
  374 |     inline void for_each(TUnaryFunction &&f)
      |                 ^~~~~~~~
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:374:17: note:   template argument deduction/substitution failed:
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:351:6: error: template argument 1 is invalid
  351 |     });
      |      ^
In file included from /home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:17:
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:390:43: note: candidate: ‘template<class TReducer, class TUnaryFunction> typename TReducer::return_type Kratos::IndexPartition<TIndexType, TMaxThreads>::for_each(TUnaryFunction&&) [with TReducer = TReducer; TUnaryFunction = TUnaryFunction; TIndexType = long unsigned int; int TMaxThreads = 128]’
  390 |     inline typename TReducer::return_type for_each(TUnaryFunction &&f)
      |                                           ^~~~~~~~
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:390:43: note:   template argument deduction/substitution failed:
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:351:6: error: template argument 1 is invalid
  351 |     });
      |      ^
In file included from /home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:17:
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:410:17: note: candidate: ‘template<class TThreadLocalStorage, class TFunction> void Kratos::IndexPartition<TIndexType, TMaxThreads>::for_each(const TThreadLocalStorage&, TFunction&&) [with TThreadLocalStorage = TThreadLocalStorage; TFunction = TFunction; TIndexType = long unsigned int; int TMaxThreads = 128]’
  410 |     inline void for_each(const TThreadLocalStorage& rThreadLocalStoragePrototype, TFunction &&f)
      |                 ^~~~~~~~
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:410:17: note:   template argument deduction/substitution failed:
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:351:6: error: template argument 1 is invalid
  351 |     });
      |      ^
In file included from /home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:17:
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:435:43: note: candidate: ‘template<class TReducer, class TThreadLocalStorage, class TFunction> typename TReducer::return_type Kratos::IndexPartition<TIndexType, TMaxThreads>::for_each(const TThreadLocalStorage&, TFunction&&) [with TReducer = TReducer; TThreadLocalStorage = TThreadLocalStorage; TFunction = TFunction; TIndexType = long unsigned int; int TMaxThreads = 128]’
  435 |     inline typename TReducer::return_type for_each(const TThreadLocalStorage& rThreadLocalStoragePrototype, TFunction &&f)
      |                                           ^~~~~~~~
/home/javi/Projects/Kratos/kratos/utilities/parallel_utilities.h:435:43: note:   template argument deduction/substitution failed:
/home/javi/Projects/Kratos/kratos/modeler/cad_tessellation_modeler.cpp:351:6: error: template argument 1 is invalid
  351 |     });
      |      ^
gmake[2]: *** [kratos/CMakeFiles/KratosCore.dir/build.make:2266: kratos/CMakeFiles/KratosCore.dir/modeler/cad_tessellation_modeler.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:715: kratos/CMakeFiles/KratosCore.dir/all] Error 2
gmake: *** [Makefile:149: all] Error 2

```

**Environment**
- OS: Ubuntu 21.04
- Branch: master

@roigcarlo

--- a/kratos/modeler/cad_tessellation_modeler.cpp
+++ b/kratos/modeler/cad_tessellation_modeler.cpp
@@ -15,6 +15,7 @@
 #include "includes/define.h"
 #include "utilities/tessellation_utilities/delaunator_utilities.h"
 #include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
 #include "cad_tessellation_modeler.h"
 
 namespace Kratos

